### PR TITLE
use safe math while finding a valid writeoff group

### PR DESCRIPTION
**Pull request by vedhavyas**

### CODE REVIEW

File: pallets/loans/src/functions.rs [hunk 1]
```diff
@@ -253,7 +254,7 @@ impl<T: Config> Pallet<T> {
 				let new_borrowed_amount = loan_info
 					.borrowed_amount
 					.checked_add(&amount)
-					.ok_or(Error::<T>::AddAmountOverflow)?;
+					.ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
```
Comment (offerijns) [inline pallets/loans/src/functions.rs:257]: I think it's a good idea to use the standard error types for this. Should we use this everywhere, i.e. also in the pools pallet? (doesn't need to be in this PR, we can just create a ticket for this)

**Review by offerijns — APPROVED** [review]
Quick turn-around, nice 🔥

File: pallets/loans/src/functions.rs (hunk 1 above)
Comment (vedhavyas) [reply pallets/loans/src/functions.rs:257]: I think we can proactively do it

Comment (branan) [reply pallets/loans/src/functions.rs:257]: I didn't realize ArithmeticError existed. We should switch to it :+1:

Comment (branan) [inline pallets/loans/src/functions.rs:257]: `DispatchError` implements `From<ArithmeticError>`. We should be able to just do `ok_or(ArithmeticError::Overflow)?` here and the `?` operator will handle the conversioe

**Pull request merged**